python_tests/test_csf.py

```diff
@@ -72,3 +75,64 @@ def test_csf_string():
         assert csf.query(key) == value
 
     os.remove(filename)
+
+
+@pytest.mark.unit
+def test_csf_load_incorrect_type_fails():
+    filename = "temp.csf"
+    with pytest.raises(caramel.CsfDeserializationException) as e:
+        keys = [f"key{i}".encode("utf-8") for i in range(1000)]
+        values = [f"value{i}" for i in range(1000)]
+        csf = caramel.CSFString(keys, values)
+        csf.save(filename)
+        csf = caramel.CSFUint32.load(filename)
+    os.remove(filename)
+
+
+@pytest.mark.unit
+def test_auto_infer_char10():
+    keys = [f"key{i}".encode("utf-8") for i in range(1000)]
+    values = [str(f"value{i}".ljust(10)[:10]) for i in range(1000)]
+    csf = caramel.CSF(keys, values)
+    assert isinstance(csf, caramel.CSFQueryWrapper)
+    filename = "temp.csf"
+    csf.save(filename)
+    csf = caramel.load(filename)
+    assert isinstance(csf, caramel.CSFQueryWrapper)
```

Comment:
should we also assert the instance of the underlying csf?

Reply:
SG, it breaks encapsulation a little to assert on `csf._csf`, but we kind of need to in order to confirm that we are picking the right backend (I'm expecting that in the future we will add many backends).